pandas/core/frame.py

```diff
@@ -393,7 +394,13 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
+    def __init__(self,
+                 data: Optional[Union[np.ndarray, dict,
+                                      ABCDataFrame]] = None,
```

Comment:
The internal ABC types evaluate to `Any` since they are created are runtime so don't help right now for static analysis. I think this should be `"DataFrame"`

Reply:
It was "DataFrame" but it looked weird to me so I changed :man_facepalming: